transcendence-app/src/user/infrastructure/db/user.entity.ts

```diff
@@ -10,30 +13,31 @@ export enum userKeys {
   AVATAR_Y = '"avatarY"',
 }
 
-type UserEntityData = {
+interface UserData {
   id: string;
   username: string;
   email: string;
   fullName: string;
   password: string | null;
-  avatarId: string | null;
+  avatarId: string;
   avatarX: number;
   avatarY: number;
   createdAt: Date;
-};
+}
 
-export class UserEntity {
+export class User {
   id: string;
   username: string;
   email: string;
   fullName: string;
+  @Exclude()
```

Comment:
I'd say all serialization may be done next to network i/o, this is, in the controllers layer as [NestJS documentations says](https://docs.nestjs.com/techniques/serialization). This entity is directly passed to some controllers but, in other cases, wrapped around a dto. In that case we'd have to serialize in two places if need to make some serialization inside de dto.

We may be still mixing a bit our approach. What do you think about unifiying like this?

- Data we **return**:
  - entities are plain data returned from DB, not serialization or logic here, just data.
  - any needed bussiness logic, mapping etc. is done in services.
  - we finally instance a dto (which has all needed serialization)
  So, with this idea in mind, we could re-use the `UserDto` for all methods returning plain users and move all serialization there.
  
- Data we **recieve**:
  - we serialize the incoming dto (we are already doing this, so good :+1: )

Reply:
Talked by call: we have arrived to an intermediate solution, please see message posted in Slack (in a few minutes)